### PR TITLE
Add a possibility to perform Shake gesture and set custom geolocation coordinates via AppleScript

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Also did some refactoring to improve the algorithm for multi-window configuration.

### CODE REVIEW

File: lib/simulator-xcode-6.js [hunk 1]
```diff
@@ -859,6 +859,22 @@ class SimulatorXcode6 extends EventEmitter {
   }
 
   /**
+   * Generate Apple script, which activates Simulator.
+   *
+   * @returns {string} the generated Apple Script snipped.
+   */
+  async generateWindowActivationScript () {
```
Comment (imurchie) [inline lib/simulator-xcode-6.js]: Why make this `async`?

Comment (imurchie) [reply lib/simulator-xcode-6.js]: Oh, I guess because a subclass needs it to be?

Comment (mykola-mokhnach) [reply lib/simulator-xcode-6.js]: yes, exactly

**Comment by imurchie** [comment]
Published in `appium-ios-simulator@2.6.0`.

**Pull request merged**